### DISCUSSION TITLE
Help require gpio/ubuntu

### DISCUSSION THREAD

**Discussion by Gazgiza** [Joshua-Riek/ubuntu-rockchip#225] [General]
Hello, I have a problem. I want to play a video file (with audio) with a script that uses the gpio ports to detect movement. Unfortunately, I have to run it in sudo but there's no sound (there is with my normal user).
Thanks for your help

**Comment by Joshua-Riek** [comment]
I looked into non-sudo access to the GPIO recently #222, this is not something I can fix. But I might have a workaround. 

Try the below commands. You might need to reboot after adding your username to the kmem group.
```
sudo usermod -aG kmem username
sudo setcap cap_sys_rawio,cap_dac_override=ep ./test
./test
```

**Reply by Gazgiza** [reply]
Oops: Unable to determine board revision from /proc/cpuinfo
 -> No "Hardware" line
 ->  You'd best google the error to find out why.
 Here is the full issue

**Reply by Gazgiza** [reply]
It's a Python script

**Reply by Joshua-Riek** [reply]
Hmmm weird, one moment, let me boot up my Opi. Btw is this the Opi 5, 5B, or 5 Plus?

**Reply by Gazgiza** [reply]
It's an Opi5

**Reply by Joshua-Riek** [reply]
Got it, does the command `gpio readall` work?

**Reply by Gazgiza** [reply]
yes

**Reply by Joshua-Riek** [reply]
Can you share your python script? Should be able to attach a text file, ill test on my side

**Reply by Gazgiza** [reply]
Failed to set capabilities on file `./test' (No such file or directory)
The value of the capability argument is not permitted for a file. Or the file is not a regular (non-symlink) file
Here is an issue when i run thee second command
Do you my script?

**Reply by Gazgiza** [reply]
import wiringpi
import time
import os
from wiringpi import GPIO

os.environ['XDG_RUNTIME_DIR'] = '/run/user/1000'

#os.system("celluloid /home/youri/Bureau/test.mp4")
#time.sleep(10000)

wiringpi.wiringPiSetup()      
wiringpi.pinMode(2, 0)       # Set pin 6 to 1 ( OUTPUT )
wiringpi.digitalWrite(2, 0)  # Write 1 ( HIGH ) to pin 6
wiringpi.digitalRead(2)      # Read pin 6

# Boucle infinie
while True:
    # Vérifier si le détecteur infrarouge détecte un mouvement
    if wiringpi.digitalRead(2) == GPIO.HIGH:
        print("Mouvement détecté !")
        os.system("mpv -ao pulse /home/youri/Bureau/vidéofinalev8.mp4")
        print("Video terminée")
        # Attendre une seconde avant de vérifier à nouveau
        time.sleep(1)

**Reply by Gazgiza** [reply]
It's in french srry

**Reply by Joshua-Riek** [reply]
No worries, ill test now.

**Reply by Gazgiza** [reply]
Don't mind the #line

**Reply by Gazgiza** [reply]
*
#os.system("celluloid /home/youri/Bureau/test.mp4")
#time.sleep(10000)

**Reply by Joshua-Riek** [reply]
The WiringOP from Orange Pi needs a lot of fixes. I can get a c program to run without sudo. But Python will give me a different error. I might need to patch the source of WiringOP to fix this. 

```bash
ubuntu@ubuntu-desktop:~$ ./foo.py 
wiringPiSetup: mmap (GPIO) failed: Operation not permitted
```

**Reply by Gazgiza** [reply]
How can I fix that?

**Reply by Gazgiza** [reply]
It's wiringOpi-Py

**Reply by Joshua-Riek** [reply]
I don't know. This looks like an issue with the libraries from Orange Pi.
